app/services/hardware/hardware.ts

```diff
@@ -31,17 +31,25 @@ export class HardwareService extends StatefulService<IHardwareServiceState> {
   }
 
   getDevices() {
-    return this.state.devices.concat(this.state.dshowDevices);
+    return this.state.devices;
   }
 
   getDevice(id: string) {
-    return this.getDevices().find(device => device.id == id);
+    return this.state.devices.find(device => device.id == id);
   }
 
   getDeviceByName(name: string) {
     return this.state.devices.find(device => device.description == name);
   }
 
+  getDshowDevices() {
+    return this.state.dshowDevices;
+  }
+
+  getDshowDevice(id: string) {
+    return this.state.dshowDevices.find(device => device.id == id);
+  }
+
```

Comment:
I would remove these if not in use as it introduces confusion on what is used or not.